crates/energy/src/battery/systems.rs

```diff
@@ -0,0 +1,23 @@
+use bevy::prelude::*;
+
+use crate::battery::component::Battery;
+use crate::battery::DISCHARGE_RATE;
+
+/// Discharges the battery of a unit.
+///
+/// # Parameters
+///
+/// - `time`: The time since the last update in seconds.
+/// - `battery`: The battery.
+pub(crate) fn discharge_battery(time: Res<Time>, mut battery: Query<&mut Battery>) {
+    for mut battery in battery.iter_mut() {
+        let energy = battery.energy();
+        if energy == 0. {
```

Comment:
`if battery.energy() == 0.` would be more concise.

Reply:
Can do, remnants of debugging prints lol.